onnx/version_converter/convert.h

```diff
@@ -611,6 +612,7 @@ class DefaultVersionConverter : public BaseVersionConverter {
     registerAdapter(std::make_unique<CompatibleAdapter>("DequantizeLinear", OpSetID(20), OpSetID(21)));
     registerAdapter(std::make_unique<CompatibleAdapter>("QuantizeLinear", OpSetID(20), OpSetID(21)));
     registerAdapter(std::make_unique<CompatibleAdapter>("QLinearMatMul", OpSetID(20), OpSetID(21)));
+    registerAdapter(std::make_unique<GroupNormalization_20_21>());
```

Comment:
It would be amazing to have a down convert adapter implemented as well. It doesn’t have to use the old group norm operator if needed. This is going to be important when we move to use the latest opset while maintaining the ability to convert back to older opsets. 


FWIW this is now we implement group norm right now: https://github.com/microsoft/onnxscript/blob/172a81745c10467e04d9b4fdfa7829ce3321b6ce/onnxscript/function_libs/torch_lib/ops/core.py#L5850

Reply:
You mean converting from opset 21 to 20? I don't think it's well defined how to convert a large tensor to a small tensor for scale and bias.

As I mentioned in the other thread, even the utility of 20->21 seems a bit dubious to me since I don't think any model should use the opset 20 definition - and I'm not aware of any.

Reply:
Could you elaborate on what you mean by convert a large tensor to a small tensor for scale and bias?

I agree that 20->21 can be less useful because very few are currently using the op. However when converters are adopting the new operator in opset 21, there will be needs for converting to lower opsets when users are trying to use runtimes that have not caught up with support.

Reply:
The difference between the two opsets is that opset20 has scale/bias shape as (num_groups,) and opset 21 has it as (C,).

The opset20 definition is in fact incorrect since it doesn't match the original paper. To go from 20 to 21, we can go through the sequence of operations defined in the adapter so that, e.g., scale=[1,2,3] becomes scale=[1,1,2,2,3,3] when C=6 and num_grps=3. However, it does not work in the other direction: suppose scale=[1,2,3,4,5,6] in opset21, we don't know how to convert it to a length-3 tensor.

Reply:
Sounds like we don’t need to use the same operator when converting back, as long as the model is correct and accurate. I think we can simply reuse the function definition (defined in this or) or refer to the implementation above (which uses instance norm) to create the opset20 adapter. Aka we don’t need to use GroupNorm. Does this address the concern?

Reply:
Happy to unblock this pr and follow up as a separate task

Reply:
https://github.com/onnx/onnx/issues/5843